src/components/DatePickerModal/readme.md

```diff
@@ -1,72 +1,312 @@
 
-initialState = { isOpen: false };
+initialState = { isOpen: false, selection: { name: '', label: '' } };
 
-function getDate(value, selectionType) {
+function getDates(value, selectionType) {
     if (selectionType === 'single' && Array.isArray(value)) return value[0];
     return value;
 }
 
-<div
-    className="rainbow-align-content_center rainbow-m-vertical_large rainbow-p-horizontal_small rainbow-m_auto rainbow-flex"
->
-    <div className="rainbow-p-horizontal_medium">
-        <Button
-            label="Click to select a date"
-            onClick={() => setState({ isOpen: true, selectionType: 'single', value: null, title: 'Select a date' })} />
-    </div>
-    <div className="rainbow-p-horizontal_medium">
-        <Button
-            label="Click to select a range of dates"
-            onClick={() => setState({ isOpen: true, selectionType: 'range', value: null, title: 'Select range of dates' })} />
-    </div>
+function setDates(value) {
+    if (value) {
+        if (value.name === 'Custom') {
+            setState({
+                isOpen: true,
+                selectionType: 'range',
+                value: null,
+                title: 'Select a Date',
+            });
+        } else if (value.name === 'Today') {
+            setState({ selection: { name: 'Today', label: 'Today' } });
+        } else if (value.name === 'Yesterday') {
+            setState({ selection: { name: 'Yesterday', label: 'Yesterday' } });
+        } else if (value.name === 'This Week') {
+            setState({ selection: { name: 'This Week', label: 'This Week' } });
+        }
+    }
+}
+
+function formatDates(date) {
+    if (date) {
+        if (date.length > 1) {
+            startDay = new Intl.DateTimeFormat().format(date[0]);
+            endDay = new Intl.DateTimeFormat().format(date[1]);
+            formatedDate = startDay + ' - ' + endDay;
+        } else {
+            formatedDate = new Intl.DateTimeFormat().format(date[0]);
+        }
+        return formatedDate;
+    }
+}
+
+const containerStyles = {
+    width: '160px',
+};
+
+<div className="rainbow-m-bottom_xx-large rainbow-p-bottom_xx-large">
+    <GlobalHeader src="images/user/user3.jpg">
+        <div className="rainbow-flex rainbow-align_right">
+            <Picklist
+                value={state.selection}
+                style={containerStyles}
+                placeholder="Select Date"
+                onChange={value => setDates(value)}
+            >
+                <Option name="Custom" label="Custom" />
+                <Option name="Today" label="Today" />
+                <Option name="Yesterday" label="Yesterday" />
+                <Option name="This Week" label="This Week" />
+            </Picklist>
+        </div>
+    </GlobalHeader>
     <DatePickerModal
         title={state.title}
         isOpen={state.isOpen}
-        value={getDate(state.value)}
+        variant="double"
+        value={getDates(state.value)}
         selectionType={state.selectionType}
```

Comment:
let's use selection type fixed like:
```selectionType="range"```
then you don't need to maintain this state also remove the getDates and you can pass value directly from state